common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.h

```diff
@@ -88,6 +89,8 @@ class base_wlan_hal_dwpal : public virtual base_wlan_hal,
     bool dwpal_nl_cmd_scan_dump();
     void *get_dwpal_nl_ctx() const { return (m_dwpal_nl_ctx); }
 
+    std::unique_ptr<nl80211_client> m_nl80211_client;
```

Comment:
What is the point of moving it to base_wlan_hal_dwpal? I though we said that it wouldn't be used by the monitor or station HALs...

More importantly though: if you move it here, it should also be taken as an argument in the base_wlan_hal_dwpal constructor and set there, not in the derived class's constructor.

Reply:
The point is that the `refresh_radio_info` is implemented in the base, so I thought that in order to use @mariomaz's #977 to read the TxPower directly from nl80211 I will need to do that from `refresh_radio_info`.
It turns out that the `TxPower` we get from hostapd is fine, but nevertheless we will probably update `refresh_radio_info` to complement the radio info parameters we get from hostapd with the values we read from nl80211, so I decided to keep it there.

Reply:
>More importantly though: if you move it here, it should also be taken as an argument in the base_wlan_hal_dwpal constructor and set there, not in the derived class's constructor.

But that's exactly what I did..

Reply:
```
base_wlan_hal_dwpal::base_wlan_hal_dwpal(HALType type, std::string iface_name,
                                         hal_event_cb_t callback, hal_conf_t hal_conf)
    : base_wlan_hal(type, iface_name, IfaceType::Intel, callback, hal_conf),
      beerocks::beerocks_fsm<dwpal_fsm_state, dwpal_fsm_event>(dwpal_fsm_state::Delay),
      m_nl80211_client(nl80211_client_factory::create_instance())
{
    LOG_IF(!m_nl80211_client, FATAL) << "Failed to create nl80211_client instance";
    // Initialize the FSM
    fsm_setup();
}
```

Reply:
I'm sorry, I misread the code.